sync-core/src/test/java/com/cloudant/sync/datastore/AttachmentTest.java

```diff
@@ -84,10 +85,18 @@ public void setBadAttachmentsTest() throws IOException, ConflictException, SQLEx
             Assert.fail("FileNotFoundException not thrown");
         } catch (FileNotFoundException fnfe) {
             // now check that things got rolled back
-            Cursor c1 = datastore.getSQLDatabase().rawQuery("select sequence from attachments;", null);
-            Assert.assertEquals("Attachments table not empty", c1.getCount(), 0);
-            Cursor c2 = datastore.getSQLDatabase().rawQuery("select sequence from revs;", null);
-            Assert.assertEquals("Revs table not empty", c2.getCount(), 1);
+            //TODO This needs to happen on the queue inside of the datastore
+            datastore.runOnDbQueue(new SQLQueueCallable<Object>() {
+                @Override
+                public Object call(SQLDatabase db) throws Exception {
+                    Cursor c1 = db.rawQuery("select sequence from attachments;", null);
+                    Assert.assertEquals("Attachments table not empty", c1.getCount(), 0);
+                    Cursor c2 = db.rawQuery("select sequence from revs;", null);
+                    Assert.assertEquals("Revs table not empty", c2.getCount(), 1);
```

Comment:
Since `ThreadLocal` variables have been removed unless we provide the test with a database connection for the main thread, we will get an `SQLiteException` thrown from the underlying database. I could change it use the `TestUtils.getDatabaseConnectionToExistingDb` to get around this, but I saw this method as a temporary stop gap until the query tests were changed.